From 2d68fa06d5cfcf38ce9bd551b478dab408141cce Mon Sep 17 00:00:00 2001
From: Mike Westerhof <mwester@dls.net>
Date: Fri, 8 Aug 2008 13:25:02 +0100
Subject: [PATCH] gta01-uart-fifo-trigger-sooner.patch

    Set the UART FIFO to trigger earlier on the GTA01 device to minimize
    UART overruns from the GSM.

    Signed-off-by: Mike Westerhof <mwester@dls.net>
---
 arch/arm/mach-s3c2410/mach-gta01.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-s3c2410/mach-gta01.c b/arch/arm/mach-s3c2410/mach-gta01.c
index 9e42df0..4927cd4 100644
--- a/arch/arm/mach-s3c2410/mach-gta01.c
+++ b/arch/arm/mach-s3c2410/mach-gta01.c
@@ -87,6 +87,8 @@ static struct map_desc gta01_iodesc[] __initdata = {
 #define UCON S3C2410_UCON_DEFAULT
 #define ULCON S3C2410_LCON_CS8 | S3C2410_LCON_PNONE | S3C2410_LCON_STOPB
 #define UFCON S3C2410_UFCON_RXTRIG8 | S3C2410_UFCON_FIFOMODE
+/* UFCON for the gta01 sets the FIFO trigger level at 4, not 8 */
+#define UFCON_GTA01_PORT0 S3C2410_UFCON_FIFOMODE
 
 static struct s3c2410_uartcfg gta01_uartcfgs[] = {
 	[0] = {
@@ -94,7 +96,7 @@ static struct s3c2410_uartcfg gta01_uartcfgs[] = {
 		.flags	     = 0,
 		.ucon	     = UCON,
 		.ulcon	     = ULCON,
-		.ufcon	     = UFCON,
+		.ufcon	     = UFCON_GTA01_PORT0,
 	},
 	[1] = {
 		.hwport	     = 1,
-- 
1.5.6.5

